### PR TITLE
Deobfuscate final files: more intelligence, default obfuscated = True, 

### PR DESCRIPTION
**Pull request by sanderjo**
plus much more unit testing

### CODE REVIEW

File: sabnzbd/deobfuscate_filenames.py [hunk 1]
```diff
@@ -97,38 +97,49 @@ def is_probably_obfuscated(myinputfilename):
...
     # little entropy in the filebasename is a sign of useless names
     if entropy(filebasename) < 3.5:
         logging.debug("Obfuscated: entropy < 3.5")
         return True
+
     # high entropy in the filebasename is a sign of useful name, so not obfuscated
     if entropy(filebasename) > 4.0:
         logging.debug("Not obfuscated: entropy > 4.0")
         return False
+    """
```
Comment (Safihre) [inline sabnzbd/deobfuscate_filenames.py:138]: If this code is not needed, please remove. For history we have Git to always recover code if we need.

**Review by Safihre — CHANGES_REQUESTED** [review]

File: sabnzbd/deobfuscate_filenames.py [hunk 2]
```diff
@@ -97,38 +97,49 @@ def is_probably_obfuscated(myinputfilename):
...
-    if spacesdots > 3:
+    # Example: "this is a download"
+    if spacesdots >= 3:
         # useful signs in filebasename, so not obfuscated
-        logging.debug("Not obfuscated: spacesdots > 3")
+        logging.debug("Not obfuscated: spacesdots >= 3")
         return False
 
-    if decimals > 3 and spacesdots > 1:
+    # Example: "Beast 2020"
```
Comment (jcfp) [inline sabnzbd/deobfuscate_filenames.py:117]: This example doesn't meet the `spacedots > 1` requirement of the check (and is only considered not obfuscated as a result of the later 'Catullus' check).

File: sabnzbd/deobfuscate_filenames.py (hunk 1 above)
Comment (sanderjo) [reply sabnzbd/deobfuscate_filenames.py:138]: I'll do!

File: sabnzbd/deobfuscate_filenames.py (hunk 2 above)
Comment (sanderjo) [reply sabnzbd/deobfuscate_filenames.py:117]: Nice find! Thanks. I'll take care of it.

**Pull request merged**